### DISCUSSION TITLE
HELP: Locked out of coolify

### DISCUSSION THREAD

**Discussion by simonjcarr** [coollabsio/coolify#3598] [Q&A] [answered]
Hi, I was having problems logging into my self hosted coolify, so I used the terminal commands to reset the root password and email address.

Now when I try to login coolify accepts the new email and password but is asking for a one time code, that I assume has been sent to my email address, however I don't remember setting up email notifications.

Do I have any options?

**Comment by fahimanwer** [comment]
any fix?

**Comment by hi2rashid** [comment]
I have the same issue
Method 1:
Set password here
/data/coolify/source/.env

![image](https://github.com/user-attachments/assets/55eda001-87db-4504-bde3-c4f1bc3ff4b9)


Method 2: 
Run the below

curl -fsSL https://cdn.coollabs.io/coolify/install.sh | bash

**Comment by castroofelipee** [comment]
Hey! I ran into the same problem where I forgot my Coolify password on a self-hosted setup. Since there’s no simple “forgot password” in the UI, I just went straight into the PostgreSQL database to reset it.

`UPDATE users SET password = '<HASH>' WHERE email = 'email@example.com';`

**Reply by cmdmasquerade** [reply]
What hash is it? Argon2? 
update: it's bcrypt

**Comment by yogiz** [comment, marked as answer]
use this to reset root password 
`docker exec -ti coolify sh -c "php artisan root:reset-password"`

it will prompt you to set new password
<img width="769" height="252" alt="image" src="https://github.com/user-attachments/assets/092573be-5302-457d-8af2-0f9823d24768" />

then login with your email and new password

**Reply by aknirob** [reply]
Thank you, brother. Best solution worked perfectly.

**Reply by manassehgitau** [reply]
This worked for me too. Learn more [here](https://coolify.io/docs/knowledge-base/commands). If you have not set up email SMTP.
**Root password reset without SMTP**
Login to your server through SSH and execute the following command:
```
docker exec -ti coolify sh -c "php artisan root:reset-password"
```
**Root email change**
Login to your server through SSH and execute the following command:
```
docker exec -ti coolify sh -c "php artisan root:change-email"
```

**Comment by redox132** [comment]
Here is the best option. You can other disable the password (not good). You can do it in here:

1. **vim /etc/ssh/sshd_config**

or
you can set the email and password of the coolify container:
check the conatiners: docker ps | grep itcontainerName
ssh into the conainter: **docker exec -itcontainerName sh**
-after that you are gonna be inside the continer. Now you can change the email and passwod using artisan command. if you do not know which command you are gonna use, use **php artisan list**. than check the related commands. Otherwise, follow up:
**php artisan root:change-email** to change the email
**php artisan root:reset-password** to change the password
you will be asked to confirm (twice input pate) the email and the password.
